tests/test_spam_admin.py

```diff
@@ -0,0 +1,440 @@
+            return (200, headers, json.dumps({"decision":"HAM"}))
+
+        httpretty.register_uri(
+            httpretty.POST, SPAM_ASSASSIN_URL,
+            body=request_callback
+        )
+
+        content = content if content is not None else 'hammer to fall'
+        url = project.api_url + 'comment/'
+        ret = self.app.post_json(
+            url,
+            {
+                'content': content,
+                'isPublic': 'public',
+                'page': 'node',
+                'target': project._id
+            },
+            **kwargs
+        )
+
+        return Comment.load(ret.json['comment']['id'])
+
+
+    ########################## TEST MODEL / UTIL FUNCTIONS  #####################################################
+
+
+    @httpretty.activate
+    def test_comment_added_is_spam(self):
+        comment = self._add_comment(
+            self.project, content = self.GTUBE, auth=self.project.creator.auth,
+        )
+        self.project.reload()
+
+
+
+        assert_equal(len(self.project.commented), 1)
+
+        if self.spam_assassin_active:
+            assert_equal(comment.spam_status, Comment.POSSIBLE_SPAM)
+        else:
+            assert_equal(comment.spam_status, Comment.UNKNOWN)
+
+    @httpretty.activate
+    def test_comment_added_is_not_spam(self):
+        comment = self._add_comment(
+            self.project,auth=self.project.creator.auth,
+        )
+        self.project.reload()
+
+
+
+        assert_equal(len(self.project.commented), 1)
+        assert_equal(comment.spam_status , Comment.UNKNOWN)
+
+    @httpretty.activate
+    def test_train_spam_comment(self):
+
+        httpretty.register_uri(
+            httpretty.POST, SPAM_ASSASSIN_TEACHING_URL,
+            body=lambda r, u, h: (200, h,json.dumps({"status":"Learned"}))
+        )
+
+        comment = self._add_comment(
+            self.project,
+            content=self.GTUBE,
+            auth=self.project.creator.auth,
+        )
+        self.project.reload()
+
+
+        if self.spam_assassin_active:
+            assert_true(train_spam(comment, is_spam=True))
+        else:
+            assert_false(train_spam(comment, is_spam=True))
+
+
+    @httpretty.activate
+    def test_train_ham_comment(self):
+
+        httpretty.register_uri(
+            httpretty.POST, SPAM_ASSASSIN_TEACHING_URL,
+            body=lambda r,u,h: (200,h,json.dumps({"status":"Learned"}))
+        )
+        comment = self._add_comment(
+            self.project, auth=self.project.creator.auth,
+        )
+        if self.spam_assassin_active:
+            assert_true(train_spam(comment, is_spam=False))
+        else:
+            assert_false(train_spam(comment, is_spam=False))
+
+    @httpretty.activate
+    def test_auto_mark_spam_if_flagged_enough_times(self):
+        comment = self._add_comment(
+            self.project, auth=self.project.creator.auth,
+        )
+
+
+        for i in range(Comment.NUM_FLAGS_FOR_SPAM):
```

Comment:
good point. The idea was to have it set to 1 so I tested it with that as well. I will write more tests with higher values for NUM_FLAGS_FOR_SPAM. I'll also have tests with the edge case of NUM_FLAGS_FOR_SPAM=1.